client/apollo/react/src/Form/Radio/CardRadio/CardRadioCommon.tsx

```diff
@@ -87,40 +82,28 @@ const CardRadioCommon = ({
         </legend>
       )}
       <div className={RadioGroupClassName}>
-        {options.map(
-          (
-            { label, description, subtitle, icon, hasError, ...inputProps },
-            index,
-          ) => (
-            <label
-              key={`${name ?? inputProps.name}`}
-              aria-invalid={Boolean(error) || hasError}
-              htmlFor={`id-${name ?? inputProps.name}-${index}`}
-              className="af-card-radio-label"
-            >
-              <RadioComponent
-                id={`id-${name ?? inputProps.name}-${index}`}
-                errorId={errorId}
-                hasError={Boolean(error) || hasError}
-                {...inputProps}
-                name={name ?? inputProps.name}
-                onChange={onChange}
-              />
-              <div className="af-card-radio-content">
-                {icon && <IconComponent src={icon} size={size} />}
-                <div className="af-card-radio-content-description">
-                  <span>{label}</span>
-                  {description && <span>{description}</span>}
-                  {subtitle && <span>{subtitle}</span>}
-                </div>
-              </div>
-            </label>
-          ),
-        )}
+        {options.map((cardRadioItemProps) => (
+          <CardRadioItem
+            key={`${name}-${crypto.randomUUID()}`}
+            name={name}
+            onChange={onChange}
+            size={size}
+            RadioComponent={RadioComponent}
+            IconComponent={IconComponent}
+            checked={
+              value !== undefined
+                ? value === cardRadioItemProps.value
+                : undefined
+            }
```

Comment:
Tu peux potentiellement simplifier comme ca ?
```suggestion
            checked={
              value !== undefined
                 && value === cardRadioItemProps.value
            }
```

Reply:
J'avais volontairement laissé undefined pour laisser la possibilité de gérer le composant de manière non contrôlée.
Avec cette proposition cela ne sera plus possible je pense.